### PR TITLE
Upgrade persistence lib 10 0 4

### PR DESCRIPTION
**Pull request by nvvi**
https://1ivy.atlassian.net/browse/MARP-1485

### CODE REVIEW

**Review by nntthuy-axonivy — COMMENTED** [review]
Hi anh @nvvi, please check my comment below. I think the code will be easier to understand if we can specify ID with Long type!

File: approval-decision-utils-test/pom.xml
[the remark this answers is not in the document]
Comment (nvvi) [reply approval-decision-utils-test/pom.xml:1]: image
the code on master is not the right format, so I correct it to readable

File: approval-decision-utils/src/com/axonivy/utils/approvaldecision/entities/BaseApprovalHistory.java [hunk 1]
```diff
@@ -9,7 +10,7 @@
 import com.axonivy.utils.persistence.beans.AuditableEntity;
 
 @MappedSuperclass
-public abstract class BaseApprovalHistory extends AuditableEntity {
+public abstract class BaseApprovalHistory<ID extends Serializable> extends AuditableEntity<ID> {
```
[the remark this answers is not in the document]
Comment (nvvi) [reply approval-decision-utils/src/com/axonivy/utils/approvaldecision/entities/BaseApprovalHistory.java:13]: in persistence lib if you want your entity extends AuditableEntity you have to provide type of your Id
you can refer to class AuditableEntity in persistence lib here
`public abstract class AuditableEntity<ID extends Serializable> extends VersionableEntity<ID> {`
this is a generic, you have to provide type of your ID.

**Review by nntthuy-axonivy — APPROVED** [review]
LGTM

**Pull request merged**